src/Analyzers/AnalyzerFormatter.cs

```diff
@@ -33,59 +36,85 @@ internal class AnalyzerFormatter : ICodeFormatter
         public async Task<Solution> FormatAsync(
             Solution solution,
             ImmutableArray<DocumentId> formattableDocuments,
-            FormatOptions options,
+            FormatOptions formatOptions,
             ILogger logger,
             List<FormattedFile> formattedFiles,
             CancellationToken cancellationToken)
         {
-            var analysisStopwatch = Stopwatch.StartNew();
-            logger.LogTrace($"Analyzing code style.");
-
-            if (!options.SaveFormattedFiles)
+            var analyzersAndFixers = _finder.GetAnalyzersAndFixers(solution, formatOptions, logger);
+            if (analyzersAndFixers.Length == 0)
             {
-                await LogDiagnosticsAsync(solution, formattableDocuments, options, logger, cancellationToken);
-            }
-            else
-            {
-                solution = await FixDiagnosticsAsync(solution, formattableDocuments, logger, cancellationToken);
+                return solution;
             }
 
+            var analysisStopwatch = Stopwatch.StartNew();
+            logger.LogTrace($"Running {_name} analysis.");
+
+            var formattablePaths = formattableDocuments.Select(id => solution.GetDocument(id)?.FilePath)
```

Comment:
Do we want to report that we could not get a file path? I thought that even though this was nullable we would expect this to be an error in `dotnet format`.

Reply:
Sure, I am not certain in practice we would ever hit that case.